### PR TITLE
LPD-31382 - tooltip does not display correct value for required multiple when a UOM and config exists

### PR DESCRIPTION
**Pull request by wanderlast**
Hi @andrea-ale-sbarra 

https://liferay.atlassian.net/browse/LPD-31382, tied to LPP. This is a fire ticket.

Per Bernard's comment [here](https://liferay.slack.com/archives/G9CMBBA9H/p1720718821538249?thread_ts=1720712439.834709&cid=G9CMBBA9H), the fix essentially tries to find the LCM of the values in the UOM and product config's multiple value.

The LCM solver as written is incredibly messy but is being done in this manner due to the potential case of _both_ the UOM and product config's values being decimals (e.g. 0.5 and 0.3 -- this can be seen in the previously written test case, in fact). Attempting to modulo them lead to the portal hanging while attempting to calculate it (which didn't happen if either of them was greater than 1 -- e.g. 5 and 0.3 worked fine as did 5 and 3). I'm unsure of why JS couldn't handle this (I'm assuming it's due to problems with floating point math), but to circumvent this since this is a fire ticket I basically attempted to find a way to make them into integers. The result is what you see below.

The previously existing tests were edited to account for the corrected tooltip value.

### CODE REVIEW

**Comment by wanderlast** [comment]
ci:test:sf

**Comment by wanderlast** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/utilities/quantities.js [hunk 1]
```diff
@@ -25,19 +25,40 @@ export function getMinQuantity(
...
+			multipleQuantity * 10 * precisionAdjustment;
 	}
 
-	//this solves for least common multiple
 	let multiple = incrementalOrderQuantity;
-	while (multiple % multipleQuantity !== 0) {
-		multiple += incrementalOrderQuantity;
+
+	while (multiple % itemMultiple !== 0) {
+		multiple += unitOfMeasureMultiple;
```
Comment (riccardo-alberti) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/utilities/quantities.js:57]: @wanderlast is it not easier to use something like this?
function findLCM(a, b) { 
    let lar = Math.max(a, b); 
    let small = Math.min(a, b); 
    for (i = lar; ; i += lar) { 
        if (i % small == 0) 
            return i; 
    } 
} 
will this not solve for all cases?

Comment (wanderlast) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/utilities/quantities.js:57]: Like I mentioned, %/modulus when used with two decimal values below 1 causes the portal to infinite loop/lag out due to it technically never reaching a valid value when evaluating i % small. You can see an example of this in the screenshot below where I used the function you provided:

Screenshot 2024-07-12 115653

Note the variable values on the left.

**Comment by wanderlast** [comment]
Changes made:
- I realized that there are some decimal values that can exist that still fall prey to the floating point math errors that are greater than 1, so I've removed that. I also improved how the precision is calculated so no precision should be lost.
- I changed the LCM logic to include the min/max calculation and the for loop per your suggestion.

I unfortunately still have not found a way for this to work without the adjustments to float values to make them integers.

**Review by riccardo-alberti — APPROVED** [review]
for the lpp it's ok like this. For master we might want to find an easier way to calculate this. thanks @wanderlast

**Pull request closed without merging**